deal/_runtime/_decorators.py

```diff
@@ -61,7 +60,7 @@ def pre(
         message=message,
         exception=exception or _exceptions.PreContractError,
     )
-    func = partial(Contracts.attach, 'pres', contract)
+    func = cast(Callable[[C], C], partial(Contracts.attach, 'pres', contract))
```

Comment:
This type casting serves no purpose. `cast` doesn't check the validity of the cast. Also, mypy now can correctly type check `partial` calls, so `cast` even reduces the type safety.

Reply:
The reason for creating these changes is mypy errors, including CI (job example: https://github.com/life4/deal/actions/runs/11742383633/job/32712993184).

You say that mypy already knows how to handle partial correctly? Then to resolve these errors, we can make a bump version of mypy?